rskj-core/src/main/java/co/rsk/peg/PegUtilsLegacy.java

```diff
@@ -0,0 +1,393 @@
+
+        // First, check tx is not a typical release tx (tx spending from any of the federation addresses and
+        // optionally sending some change to any of the federation addresses)
+        for (int i = 0; i < tx.getInputs().size(); i++) {
+            final int index = i;
+            if (activeFederations.stream().anyMatch(federation -> scriptCorrectlySpendsTx(tx, index, federation.getP2SHScript()))) {
+                return false;
+            }
+
+            if (retiredFederationP2SHScript != null && scriptCorrectlySpendsTx(tx, index, retiredFederationP2SHScript)) {
+                return false;
+            }
+
+            // Check if the registered utxo is not change from an utxo spent from either a fast bridge federation,
+            // erp federation, or even a retired fast bridge or erp federation
+            if (activations.isActive(ConsensusRule.RSKIP201)) {
+                RedeemScriptParser redeemScriptParser = RedeemScriptParserFactory.get(tx.getInput(index).getScriptSig().getChunks());
+                try {
+                    // Consider transactions that have an input with a redeem script of type P2SH ERP FED
+                    // to be "future transactions" that should not be pegins. These are gonna be considered pegouts.
+                    // This is only for backwards compatibility reasons. As soon as RSKIP353 activates,
+                    // pegins to the new federation should be valid again.
+                    // There's no reason for someone to send an actual pegin of this type before the new fed is active.
+                    // TODO: Remove this if block after RSKIP353 activation
+                    if (!activations.isActive(ConsensusRule.RSKIP353) &&
+                            (redeemScriptParser.getMultiSigType() == RedeemScriptParser.MultiSigType.P2SH_ERP_FED ||
+                                 redeemScriptParser.getMultiSigType() == RedeemScriptParser.MultiSigType.FAST_BRIDGE_P2SH_ERP_FED)) {
+                        String message = "Tried to register a transaction with a P2SH ERP federation redeem script before RSKIP353 activation";
+                        logger.warn("[isValidPegInTx] {}", message);
+                        throw new ScriptException(message);
+                    }
+                    Script inputStandardRedeemScript = redeemScriptParser.extractStandardRedeemScript();
+                    if (activeFederations.stream().anyMatch(federation -> federation.getStandardRedeemScript().equals(inputStandardRedeemScript))) {
+                        return false;
+                    }
+
+                    Script outputScript = ScriptBuilder.createP2SHOutputScript(inputStandardRedeemScript);
+                    if (outputScript.equals(retiredFederationP2SHScript)) {
+                        return false;
+                    }
+                } catch (ScriptException e) {
+                    // There is no redeem script, could be a peg-in from a P2PKH address
```

Comment:
Maybe Add a Log Here?

Reply:
Legacy code, nothing new will be executed here so not really use for a log